src/include/kernel/mem/vmm.h

```diff
@@ -11,13 +13,38 @@
 #define PHYS_ADDRESS_MAPPED 0b1
 #define PHYS_ADDRESS_MISMATCH 0b10 // This is returned when given a phys and virt address, the virt address does not contain the phys one
 
+#define VIRT_ADDRESS_PRESENT 1
+#define VIRT_ADDRESS_NOT_PRESENT 0
+
+#define VM_KERNEL_MEMORY_PADDING PAGE_SIZE_IN_BYTES
+
+#define VMM_RESERVED_SPACE_SIZE 0x14480000000
+
 #define VPTR(x) (void*)((uint64_t)(x))
 
+typedef struct VmmItem{
+    uintptr_t base;
+    size_t size;
+    size_t flags;
+} VmmItem;
+
+typedef struct VmmContainer {
+    VmmItem vmm_root[(PAGE_SIZE_IN_BYTES/sizeof(VmmItem)) - 1];
+    struct VmmContainer *next;
+} __attribute__((__packed__)) VmmContainer;
+
+extern uint64_t end_of_vmm_space;
+
+void vmm_init();
+
+void *vmm_alloc();
+
 void *map_vaddress(void *, unsigned int);
 int unmap_vaddress(void *);
 
 void *map_phys_to_virt_addr(void* physical_address, void* address, unsigned int flags);
 void identity_map_phys_address(void *pyhysical_address, unsigned int flags);
-void map_vaddress_range(void *, unsigned int, size_t);
-uint8_t is_phyisical_address_mapped(uint64_t, uint64_t);
+void map_vaddress_range(void *virtual_address, unsigned int flags, size_t required_pages);
+uint8_t is_phyisical_address_mapped(uint64_t physical_address, uint64_t virtual_address);
```

Comment:
`uint64_t` is fine for 64bit platform, but it may better to use `uintptr_t` to indicate that 'this is a native cpu word sized address' - and not really care about how many bits there are.